src/coffee/run.coffee

```diff
@@ -4,15 +4,20 @@ fs = Promise.promisifyAll require('fs')
 Csv = require 'csv'
 JSZip = require 'jszip'
 ProductTypeGenerator = require './product-type-generator'
+{SphereClient} = require 'sphere-node-sdk'
+package_json = require '../package.json'
 
 argv = require('optimist')
-  .usage('Usage: $0 --types [CSV] --attributes [CSV] --target [folder] --retailer --zip --zipFileName [name]')
+  .usage('Usage: $0 --types [CSV] --attributes [CSV] --target [folder] --retailer --zip --zipFileName [name] --projectKey [key] --clientId [id] --clientSecret [secret]')
   .describe('types', 'path to CSV file describing product-type general info')
   .describe('attributes', 'path to CSV file describing product-type attributes info')
   .describe('target', 'target directory for generated product types JSON files')
   .describe('withRetailer', 'whether to generate an extra file for master<->retailer support with a "mastersku" attribute or not')
   .describe('zip', 'whether to zip the target folder or not')
   .describe('zipFileName', 'the zipped file name (without extension)')
+  .describe('projectKey', 'your SPHERE.IO project-key')
```

Comment:
because for nested types we need to find out the internal ID of a product type to create the correct JSON, in absence of an external key for now the name is used for reference